error message when debugging fails to start does not work anymore

previously, when the WebSocket protocol is not enabled on the webserver and debugging is started we get the error-message:

_"Failed to start debugging session. Check that Intersystems server webserver supports WebSocktets"_

implemented here:

[ Improve error message when debugging fails to start #908 ](https://github.com/intersystems-community/vscode-objectscript/pull/908)

**Now, it seems this is broken in the current version, no message will be pop'ed up, nothing happens, debugging is just not working.**

If you look in the dev-tools, the following error is shown in the console log:

mainThreadExtensionService.ts:78 [intersystems-community.vscode-objectscript]Unexpected server response: 101
$onExtensionRuntimeError	@	mainThreadExtensionService.ts:78
S	@	rpcProtocol.ts:458
Q	@	rpcProtocol.ts:443
M	@	rpcProtocol.ts:373
L	@	rpcProtocol.ts:299
(anonymous)	@	rpcProtocol.ts:161
B	@	event.ts:1208
fire	@	event.ts:1239
fire	@	ipc.net.ts:652
l.onmessage	@	localProcessExtensionHost.ts:378

mainThreadExtensionService.ts:79 Error: Unexpected server response: 101
	at ClientRequest.<anonymous> (c:\Users\<user>\.vscode\extensions\intersystems-community.vscode-objectscript-2.12.10\dist\extension.js:2:811296)
	at ClientRequest.emit (node:events:518:28)
	at HTTPParser.parserOnIncomingClient [as onIncoming] (node:_http_client:702:27)
	at HTTPParser.parserOnHeadersComplete (node:_http_common:118:17)
	at Socket.socketOnData (node:_http_client:544:22)
	at Socket.emit (node:events:518:28)
	at addChunk (node:internal/streams/readable:561:12)
	at readableAddChunkPushByteMode (node:internal/streams/readable:512:3)
	at Readable.push (node:internal/streams/readable:392:5)
	at TCP.onStreamRead (node:internal/stream_base_commons:191:23)
$onExtensionRuntimeError	@	mainThreadExtensionService.ts:79
S	@	rpcProtocol.ts:458
Q	@	rpcProtocol.ts:443
M	@	rpcProtocol.ts:373
L	@	rpcProtocol.ts:299
(anonymous)	@	rpcProtocol.ts:161
B	@	event.ts:1208
fire	@	event.ts:1239
fire	@	ipc.net.ts:652
l.onmessage	@	localProcessExtensionHost.ts:378

log.ts:456   ERR Ein unbekannter Fehler ist aufgetreten. Weitere Details dazu finden Sie im Protokoll.
error	@	log.ts:456
error	@	log.ts:561
error	@	logService.ts:51
kc	@	workbench.ts:116
(anonymous)	@	workbench.ts:97
onUnexpectedError	@	errors.ts:65
mt	@	errors.ts:110
$onUnexpectedError	@	mainThreadErrors.ts:21
S	@	rpcProtocol.ts:458
Q	@	rpcProtocol.ts:443
M	@	rpcProtocol.ts:373
L	@	rpcProtocol.ts:299
(anonymous)	@	rpcProtocol.ts:161
B	@	event.ts:1208
fire	@	event.ts:1239
fire	@	ipc.net.ts:652
l.onmessage	@	localProcessExtensionHost.ts:378

In case the websockets connection does not work or is not enabled on the webserver, the UI should show an error-message. (as it works before)

Thx.

--- a/src/debug/debugSession.ts
+++ b/src/debug/debugSession.ts
@@ -214,6 +214,7 @@ export class ObjectScriptDebugSession extends LoggingDebugSession {
           this.sendEvent(new OutputEvent(warning + "\n"));
         })
         .on("close", disposeConnection)
+        .on("error", disposeConnection)
         .on("stdout", (data: string) => {
           this.sendEvent(new OutputEvent(data, "stdout"));
         });

